src/nnet3/decodable-batch-looped.cc

```diff
@@ -0,0 +1,620 @@
+// nnet3/decodable-batch-looped.cc
+
+// Copyright      2020  XiaoMi.corp (author: yanzhao)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#if HAVE_CUDA == 1
+#include "nnet3/decodable-batch-looped.h"
+#include "nnet3/nnet-utils.h"
+#include "nnet3/nnet-compile-looped.h"
+#include "base/timer.h"
+
+namespace kaldi {
+namespace nnet3 {
+
+
+DecodableNnetBatchLoopedInfo::DecodableNnetBatchLoopedInfo(
+    const NnetBatchLoopedComputationOptions &opts,
+    Nnet *nnet):
+    opts(opts), nnet(*nnet) {
+  Init(opts, nnet);
```

Comment:
Since `this` contains `opts` and `nnet` and `Init` is not a static method, is it possible to make `Init` accept no arguments at all?

That is, use
```cpp
  void Init();
```

Reply:
yes, it is. But it's not nessary.